addons/point_of_sale/models/pos_session.py

```diff
@@ -380,7 +380,10 @@ def _accumulate_amounts(self, data):
 
             if order.is_invoiced:
                 # Combine invoice receivable lines
-                key = order.partner_id.property_account_receivable_id.id
+                if order.payment_ids[:1].payment_method_id.split_transactions:
+                    key = (order.partner_id.property_account_receivable_id.id, payment)
+                else:
+                    key = order.partner_id.property_account_receivable_id.id
```

Comment:
Keep the key type consistent:
```suggestion
                    key = (order.partner_id.property_account_receivable_id.id, False)
```